casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -198,8 +208,8 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Capture: ConnectionsCell: Tr
 
   def addDeploy(deploy: Deploy): F[Unit] =
     for {
-      _ <- Sync[F].delay {
-            deployHist += deploy
+      _ <- Cell[F, CasperState].modify { s =>
+            s.copy(deployHistory = s.deployHistory + deploy).pure[F]
```

Comment:
maybe it would make sense to create a method like `def modify(f: S => S): F[Unit]` in Cell?

Reply:
I could have a go in a follow up PR